cmake/recipes/mkl.cmake

```diff
@@ -400,6 +400,10 @@ if(NOT MSVC)
     target_link_libraries(mkl::mkl INTERFACE ${LIBM_LIBRARY})
 endif()
 
+if(NOT MSVC)
+    target_link_libraries(mkl::mkl INTERFACE ${CMAKE_DL_LIBS})
```

Comment:
Why is this needed specifically on the HPC?

Reply:
Idk exactly, but if I don't have it linker errors:

```
/usr/bin/ld: ../_deps/mkl-static-src/lib/libmkl_core.a(mkl_memory_patched.o): undefined reference to symbol 'dlclose@@GLIBC_2.2.5'
//usr/lib64/libdl.so.2: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
```

I am using `gcc (GCC) 10.2.0`.

Reply:
I don't get this linker error on any of our other Linux machines.

Reply:
I also get a lot of linker errors from MKL when building TBB statically (and using the TBB backend for MKL).

```
../_deps/mkl-static-src/lib/libmkl_tbb_thread.a(zpotrf_tbb.o): In function `mkl_lapack_zpotrf':
zpotrf_tbb.cpp:(.text+0x3fe): undefined reference to `tbb::detail::r1::initialize(tbb::detail::d1::task_group_context&)'
zpotrf_tbb.cpp:(.text+0x41a): undefined reference to `tbb::detail::r1::notify_waiters(unsigned long)'
zpotrf_tbb.cpp:(.text+0x42c): undefined reference to `tbb::detail::r1::execute_and_wait(tbb::detail::d1::task&, tbb::detail::d1::task_group_context&, tbb::detail::d1::wait_context&, tbb::detail::d1::task_group_context&)'
zpotrf_tbb.cpp:(.text+0x435): undefined reference to `tbb::detail::r1::destroy(tbb::detail::d1::task_group_context&)'
```

@jdumas, have you encountered this issue before?

Reply:
Not sure what's causing the dlclose issue on some Linux flavor and not others (maybe an old version of glibc?). For the TBB issue, I think we've fixed it by updating our TBB version to v2021.6.0-rc1 (you can copy the recipe from Lagrange again).